### PR TITLE
replace data init function with consts for typeinfov2

### PR DESCRIPTION
**Pull request by ringabout**
fixes https://forum.nim-lang.org/t/9676

### CODE REVIEW

File: compiler/ccgtypes.nim [hunk 1]
```diff
@@ -1350,17 +1350,16 @@ proc genDisplay(t: PType, depth: int): Rope =
 
 proc genTypeInfoV2Impl(m: BModule; t, origType: PType, name: Rope; info: TLineInfo) =
   cgsym(m, "TNimTypeV2")
-  m.s[cfsStrData].addf("N_LIB_PRIVATE TNimTypeV2 $1;$n", [name])
+  m.s[cfsStrData].addf("extern N_LIB_PRIVATE TNimTypeV2 $1;$n", [name])
```
Comment (ringabout) [inline compiler/ccgtypes.nim:1353]: C++ needs `extern`, see https://stackoverflow.com/questions/3714808/forward-declarations-for-variables

File: compiler/ccgtypes.nim [hunk 2]
```diff
@@ -1350,17 +1350,19 @@ proc genDisplay(t: PType, depth: int): Rope =
 
 proc genTypeInfoV2Impl(m: BModule; t, origType: PType, name: Rope; info: TLineInfo) =
   cgsym(m, "TNimTypeV2")
-  m.s[cfsStrData].addf("N_LIB_PRIVATE TNimTypeV2 $1;$n", [name])
+  if m.compileToCpp: # todo how to handle C++?
```
Comment (ringabout) [inline compiler/ccgtypes.nim]: C++ doesn't support forward variables without external linkage, which is not desirable. Probably, we should generate the `preMain` initialization for `--nomain:off` option.

ref https://stackoverflow.com/questions/3714808/forward-declarations-for-variables

Comment (ringabout) [reply compiler/ccgtypes.nim]: How about keeping the original code for C++, I can improve C++ codegen in the following PRs.

Comment (Araq) [reply compiler/ccgtypes.nim]: Sounds good.

**Comment by ringabout** [comment]
I checked locally. It works for https://forum.nim-lang.org/t/9676

**Pull request merged**